### PR TITLE
AEA-1458 fixes for flaky tests

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

fixes for flaky tests

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes and CI passes too
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: pytest.ini [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [pytest]
 log_cli = 1
-log_cli_level = DEBUG
+log_cli_level = CRITICAL
```
Comment (solarw) [inline pytest.ini:3]: this one reduce live output  of python logging by pytest
logging output will be hidden till test fails

File: tests/test_helpers/test_logging.py [hunk 2]
```diff
@@ -36,6 +36,7 @@ def test_agent_logger_adapter():
     """Test the agent logger adapter."""
     logger = logging.getLogger("some.logger")
     logger = AgentLoggerAdapter(logger, agent_name="some_agent")
+    logger.setLevel("DEBUG")
```
Comment (DavidMinarsch) [inline tests/test_helpers/test_logging.py:39]: Keep?

File: tests/test_launcher.py [hunk 3]
```diff
@@ -77,14 +77,31 @@ def set_runtime_mode_to_async(cls, agent_name: str) -> None:
...
             )
             runner.start(True)
             wait_for_condition(lambda: runner.is_running, timeout=10)
+
+            capfd_out = ""
+
+            def _check():
+                nonlocal capfd_out
+                capfd_out += capfd.readouterr().out
+                log_text = capfd_out + "\n".join(caplog.messages)
```
Comment (DavidMinarsch) [inline tests/test_launcher.py:96]: Add explanation here why both used.

File: tests/test_helpers/test_logging.py (hunk 2 above)
Comment (solarw) [reply tests/test_helpers/test_logging.py:39]: yes

**Review by marcofavorito — APPROVED** [review]
LGTM

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**